src/main/java/org/broadinstitute/hellbender/utils/bigquery/BigQueryUtils.java

```diff
@@ -0,0 +1,449 @@
+        Job   queryJob       = bigQuery.create(JobInfo.newBuilder(queryJobConfiguration).setJobId(jobId).build());
+
+        // Wait for the query to complete.
+        try {
+            logger.info("Waiting for query to complete...");
+            queryJob = queryJob.waitFor();
+        }
+        catch (final InterruptedException ex) {
+            throw new GATKException("Interrupted while waiting for query job to complete", ex);
+        }
+
+        // Check for errors:
+        if (queryJob == null) {
+            throw new GATKException("Query job no longer exists");
+        } else if (queryJob.getStatus().getError() != null) {
+
+            // Get all the errors we found and log them:
+            for ( final BigQueryError bigQueryError : queryJob.getStatus().getExecutionErrors() ) {
+                logger.error( "Encountered BigQuery Execution Error: " + bigQueryError.toString() );
+            }
+
+            // Since we found an error, we should stop and alert the user:
+            throw new GATKException(queryJob.getStatus().getError().toString());
+        }
+
+        // Get the results.
+        logger.info("Retrieving query results...");
+        final QueryResponse response = bigQuery.getQueryResults(jobId);
+        final TableResult result;
+        try {
+            result = queryJob.getQueryResults();
+
+            long bytesProcessed = ((JobStatistics.QueryStatistics) queryJob.getStatistics()).getTotalBytesProcessed();
+            logger.info(String.format("Actual %s MB scanned", bytesProcessed/1000000));
+
+        }
+        catch (final InterruptedException ex) {
+            throw new GATKException("Interrupted while waiting for query job to complete", ex);
+        }
+
+        return result;
+    }
+
+    private static long getQueryCostBytesProcessedEstimate(String queryString) {
+        final QueryJobConfiguration dryRunQueryConfig =
+                QueryJobConfiguration.newBuilder( queryString )
+                        .setUseLegacySql(false)
+                        .setDryRun(true)
+                        .setUseQueryCache(false)
+                        .setPriority(QueryJobConfiguration.Priority.INTERACTIVE)
+                        .build();
+
+        Job dryRunJob = getBigQueryEndPoint().create(JobInfo.newBuilder(dryRunQueryConfig).build());
+        long bytesProcessed = ((JobStatistics.QueryStatistics) dryRunJob.getStatistics()).getTotalBytesProcessed();
+        return bytesProcessed;
+    }
+
+//    public static StorageAPIAvroReader executeQueryWithStorageAPI(final String queryString, final List<String> fieldsToRetrieve, final String projectID) {
+//
+//        return executeQueryWithStorageAPI(queryString, fieldsToRetrieve, projectID, false);
+//    }
+//
+//    public static StorageAPIAvroReader executeQueryWithStorageAPI(final String queryString, final List<String> fieldsToRetrieve, final String projectID, final boolean runQueryInBatchMode) {
```

Comment:
You include the `StorageAPIAvroReader` in this branch, and yet you commented out the `BigQueryUtils` methods that actually use it. Can you explain? As I'm sure you know, the Storage API provides much better performance than the regular API, so it seems useful to have these utilities here. Are you going to be using the Storage API in your subsequent PRs?